Tests/MapboxMapsTests/Foundation/MapboxObservableTests.swift

```diff
@@ -270,6 +469,56 @@ final class MapboxObservableTests: XCTestCase {
             Set(unsubscribedObservers.map(ObjectIdentifier.init)))
     }
 
+    func testTypedUnsubscribesOnDeinit() throws {
+        func verifyInvocation<Payload>(
+            eventType: MapEvents.Event<Payload>,
+            handlerStub: Stub<MapEvent<Payload>, Void> = .init()
+        ) throws {
+            let otherObserver = MockObserver()
+            let subscribedObservers: [Observer]
+
+            do {
+                let mapboxObservable = MapboxObservable(observable: observable)
+                mapboxObservable.subscribe(observer, events: events)
+                mapboxObservable.subscribe(otherObserver, events: events)
+                _ = mapboxObservable.onNext(event: eventType, handler: handlerStub.call(with:))
+                _ = mapboxObservable.onEvery(event: eventType, handler: handlerStub.call(with:))
+
+                XCTAssertEqual(observable.subscribeStub.invocations.count, 4)
+                subscribedObservers = observable.subscribeStub.invocations.map(\.parameters.observer)
+            }
+
+            XCTAssertEqual(observable.unsubscribeStub.invocations.count, 4)
+            let unsubscribedObservers = observable.unsubscribeStub.invocations.map(\.parameters)
+
+            XCTAssertEqual(
+                Set(subscribedObservers.map(ObjectIdentifier.init)),
+                Set(unsubscribedObservers.map(ObjectIdentifier.init)))
+        }
+
+        // swiftlint:disable opening_brace
+        let eventInvocations = [
+            { try verifyInvocation(eventType: .mapLoaded) },
+            { try verifyInvocation(eventType: .mapLoadingError) },
+            { try verifyInvocation(eventType: .mapIdle) },
+            { try verifyInvocation(eventType: .styleDataLoaded) },
+            { try verifyInvocation(eventType: .styleLoaded) },
+            { try verifyInvocation(eventType: .styleImageMissing) },
+            { try verifyInvocation(eventType: .styleImageRemoveUnused) },
+            { try verifyInvocation(eventType: .sourceDataLoaded) },
+            { try verifyInvocation(eventType: .sourceAdded) },
+            { try verifyInvocation(eventType: .sourceRemoved) },
+            { try verifyInvocation(eventType: .renderFrameStarted) },
+            { try verifyInvocation(eventType: .renderFrameFinished) },
+            { try verifyInvocation(eventType: .cameraChanged) },
+            { try verifyInvocation(eventType: .resourceRequest) }
+        ]
+        // swiftlint:enable opening_brace
+
+        try eventInvocations.randomElement()!()
```

Comment:
Why do we go for a random one instead of verifying each of them?

Reply:
1. This is how the `EventKind` events are tested
2. It's matter of balance on the scale time<->test coverage and this provides a good middle ground

Reply:
How long this test takes?

Reply:
0.1s - random element vs 0.36s for all. There are 11 tests like this.